src/shared/directives/conditional-show/domain/conditional-show.model.ts

```diff
@@ -126,7 +126,8 @@ export class ShowCondition {
             try {
               let arrayIndex = Number.parseInt(pathTail[0], 10);
               const [__, ...dropNumberPath] = pathTail;
-              return this.findValueForComplexCondition(fields[head][arrayIndex]['value'], tail[0], tail.slice(1), dropNumberPath.join('_'));
+              let newFields = fields[head][arrayIndex] !== undefined ? fields[head][arrayIndex]['value'] : fields[head];
```

Comment:
Shouldn't we check why this error is happening and fix the cause? Current solution is just making incorrect data not throwing the error.

Reply:
A click on remove button in turn calls Angular `formArray.removeAt(index)` which is emitting  valueChanges and statusChanges events.  I'm not sure whether we can control these in case of remove only? Talk to you later on this as my UI knowledge is very limited.
https://github.com/hmcts/ccd-case-ui-toolkit/blob/RDM-4121/src/shared/components/palette/collection/write-collection-field.component.ts#L121

In any case above safeValue check helps to deal with bad data rather than blow up.